src/gt4py/analysis/passes.py

```diff
@@ -865,3 +865,133 @@ def _make_axis_interval(self, interval: IntervalInfo):
         result = gt_ir.AxisInterval(start=axis_bounds[0], end=axis_bounds[1])
 
         return result
+
+
+class DemoteLocalTemporariesToVariablesPass(TransformPass):
```

Comment:
I think we should add the typing hints for the return values, at least in the new functions/methods